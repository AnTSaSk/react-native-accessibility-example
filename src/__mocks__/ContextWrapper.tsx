import React, { FunctionComponent } from 'react';
import { AppearanceProvider } from 'react-native-appearance/src/mock';

// Context
import { AccessibilityProvider } from '../hooks/Accessibility';
import { ThemeProvider } from '../hooks/Theme';

const ContextWrapper: FunctionComponent<{}> = ({ children }) => (
  <AppearanceProvider>
    <AccessibilityProvider>
      <ThemeProvider>{children}</ThemeProvider>
    </AccessibilityProvider>
  </AppearanceProvider>
);

export default ContextWrapper;
