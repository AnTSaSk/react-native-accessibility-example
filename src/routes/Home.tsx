import React, { FunctionComponent } from 'react';
import { SafeAreaView, View } from 'react-native';

// Components
import Title from '../components/Title';
import Paragraph from '../components/Paragraph';

const Home: FunctionComponent = () => {
  return (
    <>
      <SafeAreaView>
        <View>
          <View>
            <Title label="Titre de l'article - Autisme" />
            <Paragraph>
              Voici un exemple de paragraphe limité en nombre de caractères/mots
              par ligne, allant à l&apos;essentiel et est donc plus agréable
              pour un utilisateur autiste Asperger.
            </Paragraph>
            <Paragraph>
              Cet autre paragraphe aborde le même sujet mais n&apos;as pas de
              nécessité à faire partir du premier, rendant ainsi la lecture plus
              agréable et conçise.
            </Paragraph>
          </View>
        </View>
      </SafeAreaView>
    </>
  );
};

export default Home;
