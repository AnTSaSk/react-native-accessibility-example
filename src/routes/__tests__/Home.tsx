import React, { FunctionComponent } from 'react';
import renderer from 'react-test-renderer';

// Mocked context
import ContextWrapper from '../../__mocks__/ContextWrapper';

// Route
import Home from '../Home';

const TestComponent: FunctionComponent<{}> = () => (
  <ContextWrapper>
    <Home />
  </ContextWrapper>
);

describe('Route - Home', () => {
  const instance = renderer.create(<TestComponent />);

  it('Renders correctly component', () => {
    const JSON = instance.toJSON();

    expect(JSON).toMatchSnapshot();
  });
});
