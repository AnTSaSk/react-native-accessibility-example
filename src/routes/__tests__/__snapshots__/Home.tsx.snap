// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route - Home Renders correctly component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      style={
        Object {
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        }
      }
    >
      <Text
        accessibilityLabel="Titre de l'article - Autisme"
        accessibilityRole="text"
        accessible={true}
        isDark={false}
        label="Titre de l'article - Autisme"
        style={
          Object {
            "color": "#A9A9A9",
            "fontSize": 26,
            "marginBottom": 15,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
          }
        }
      >
        Titre de l'article - Autisme
      </Text>
      <Text
        accessibilityRole="text"
        accessible={true}
        isDark={false}
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          }
        }
      >
        Voici un exemple de paragraphe limité en nombre de caractères/mots par ligne, allant à l'essentiel et est donc plus agréable pour un utilisateur autiste Asperger.
      </Text>
      <Text
        accessibilityRole="text"
        accessible={true}
        isDark={false}
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          }
        }
      >
        Cet autre paragraphe aborde le même sujet mais n'as pas de nécessité à faire partir du premier, rendant ainsi la lecture plus agréable et conçise.
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
